### DISCUSSION TITLE
Non-Revenue KPI - Setting Priors

### DISCUSSION THREAD

**Discussion by brettackers** [google/meridian#257] [General]
Hi All,

We're currently running our first Meridian model using our own data, and we've hit a roadblock when configuring the model's priors. We have a non-revenue KPI, and as we're not able to tie revenue per KPI to each channel, we're after some guidance as to how to best set our priors.

Looking at the [documentation](https://developers.google.com/meridian/docs/advanced-modelling/unknown-revenue-kpi), given we do not have past experiment results available, it is suggested that we set a channel-level contribution priors, which would be based on intuition. 

Is there a more calculated method for how we best set the priors, or have a recommended approach? Our model currently has 28 paid media channels, including both digital (SEM, display, video) and offline (TV, radio, OOH).

Please let me know if you require any more information.

**Comment by kgierach-hf** [comment]
Hi Brett,

For setting of priors, I would recommend using this general methodology described in [Steven Miller's blog post](https://svmiller.com/blog/2021/02/thinking-about-your-priors-bayesian-analysis/).  Another option is to try to set your initial priors as non-informative as possible, mostly as guard rails to keep the model out of trouble, and let the model learn from the data initially; only introducing refinements iteratively.

I suspect you will need to combine channels into groups, as the model is linear, and 28 degrees of freedom will be too much in my experience, to accurately model the effects given the linear nature of the model.

**Reply by brettackers** [reply]
thank you!

**Comment by yingxzgo** [comment]
+1 to combining channels into groups.

Another option is to set the total media contribution priors if channel-level priors are challenging to determine. You can find more details in (https://developers.google.com/meridian/docs/advanced-modeling/unknown-revenue-kpi#set-total-media-contribution-prior).

To do this, start by forming a hypothesis about the proportion of the KPI that’s incremental due to all 28 paid media channels combined. Based on this intuition, you can then assign a mean and standard deviation for this proportion. The code snippet linked above calculates a log-normal 'ROI prior' that can be applied to all channels and will result in the total contribution proportion having the desired prior mean and standard deviation that you set. (Here, ROI is defined as KPI units per spend unit). The model will then use the data to differentiate channel performance and refine the posterior ROI estimates accordingly.

**Reply by paulina-hf** [reply]
When calculating the ROI prior (for total media contribution), do I need to scale my spend variables? So far we have not been doing anything to the spend variables, and the model results in a negative baseline.

**Comment by gousesb** [comment]
Hello @paulina-hf ,

Thank you for contacting us!

- See [Custom total paid media contribution prior](https://developers.google.com/meridian/docs/advanced-modeling/unknown-revenue-kpi-custom#set-total-paid-media-contribution-prior) for the calculation for setting a total media contribution prior. Beyond this calculation, no additional scaling is needed. See [Input data](https://developers.google.com/meridian/docs/basics/input-data) for transformations Meridian automatically applies.
- See [Assess the baseline](https://developers.google.com/meridian/docs/advanced-modeling/baseline) for strategies around mitigating negative baseline.

Feel free to reach out if you have any further queries or feedback regarding Meridian.

Thank you

Google Meridian Support Team
